### ISSUE TITLE
translate-c: error: use of undeclared identifier 'struct_name'

### ISSUE THREAD

**Issue by Ristovski** [ziglang/zig#13468]
### Zig Version

0.11.0-dev.85+694d8831c

### Steps to Reproduce and Observed Behavior

`zig translate-c` fails to translate a fairly common macro found in many headers:
```C
#define STRUCTIFY(name) struct name
```

```
$ zig translate-c test.h > test.zig && zig ast-check test.zig
test.zig:471:48: error: use of undeclared identifier 'struct_name'
pub inline fn STRUCTIFY(name: anytype) @TypeOf(struct_name) {
                                               ^~~~~~~~~~~
```

The translated output is the following:
```Zig
pub inline fn STRUCTIFY(name: anytype) @TypeOf(struct_name) {
    _ = @TypeOf(name);
    return struct_name;
}
```

### Expected Behavior

Both `zig translate-c` and `@cInclude`'ing the header should work.

**Comment by kassane** [comment]
This is a similar problem I had using [`libxlsxwriter`](https://github.com/jmcnamara/libxlsxwriter).

This code block isn't translated. It causes error in other operations - on `cimport.zig`:
```c
#define	TAILQ_HEAD(name, type)						\
struct name {								\
	struct type *tqh_first;	/* first element */			\
	struct type **tqh_last;	/* addr of last next element */		\
	TRACEBUF							\
}
```
full-file: 
https://github.com/jmcnamara/libxlsxwriter/blob/43037c1e883e3b2fcb5ea4aaf6cfaf46bf434b76/include/xlsxwriter/third_party/queue.h#L491-L693


## Expected

Build works

## Fail

```bash
zig build             
/home/kassane/Documentos/Projetos/zig_xlsxwriter/zig-cache/o/e9de012ef0a7dd1c616dc3f6012821a0/cimport.zig:4706:110: error: use of undeclared identifier 'struct_headname'
pub inline fn TAILQ_LAST(head: anytype, headname: anytype) @TypeOf(@import("std").zig.c_translation.cast([*c]struct_headname, head.*.tqh_last).*.tqh_last.*) {
                                                                                                             ^~~~~~~~~~~~~~~
/home/kassane/Documentos/Projetos/zig_xlsxwriter/zig-cache/o/e9de012ef0a7dd1c616dc3f6012821a0/cimport.zig:4714:125: error: use of undeclared identifier 'struct_headname'
pub inline fn TAILQ_PREV(elm: anytype, headname: anytype, field: anytype) @TypeOf(@import("std").zig.c_translation.cast([*c]struct_headname, elm.*.field.tqe_prev).*.tqh_last.*) {
                                                                                                                            ^~~~~~~~~~~~~~~
/home/kassane/Documentos/Projetos/zig_xlsxwriter/examples/tutorial1.zig:21:17: error: expected type '[32]u8', found '*const [4:0]u8'
        .item = "Rent",
                ^~~~~~
/home/kassane/Documentos/Projetos/zig_xlsxwriter/examples/tutorial1.zig:1:20: error: C import failed: AnalysisFail
const xlsxwriter = @cImport(@cInclude("xlsxwriter.h"));
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: tutorial1...
```
---

## Test

**C** src: https://github.com/jmcnamara/libxlsxwriter/blob/main/examples/tutorial1.c

**Zig**

```zig
const xlsxwriter = @cImport(@cInclude("xlsxwriter.h"));

const Expense = struct {
    item: [32]u8,
    cost: u32,
};

 var expenses = [_]Expense{
    .{ .item = "Rent", .cost = 1000 },
    .{ .item = "Gas", .cost = 100 },
    .{ .item = "Food", .cost = 300 },
    .{ .item = "Gym", .cost = 50 },
};


pub fn main() void {

    // Create a workbook and add a worksheet.
    const workbook: *xlsxwriter.lxw_workbook = xlsxwriter.workbook_new("tutorial1.xlsx");
    const worksheet: *xlsxwriter.lxw_worksheet = xlsxwriter.workbook_add_worksheet(workbook, 0);

    var row: c_int = 0;
    var col: c_int = 0;
    {

   // Iterate over the data and write it out element by element.
        while (row < 4) : (row +=1)  {
             _ =  xlsxwriter.worksheet_write_string(worksheet, row, col, expenses[row].item, null);
             _ =  xlsxwriter.worksheet_write_number(worksheet, row, col + 1, expenses[row].cost, null);
         }
   }

    // Write a total using a formula.
    _ = xlsxwriter.worksheet_write_string(worksheet, row, col, "Total", 0);
    _ = xlsxwriter.worksheet_write_formula(worksheet, row, col + 1, "=SUM(B1:B4)", 0);

    // Save the workbook and free any allocated memory.
    _ = xlsxwriter.workbook_close(workbook);
}
```

**Comment by valpackett** [comment]
This happens with any FreeBSD system header that pulls `queue.h` :(

```
zig-cache/o/a6f5dfe6a9d692c7d0fdbc4ef13c56d8/cimport.zig:918:54: error: use of undeclared identifier 'struct_type'
pub inline fn QUEUE_TYPEOF(@"type": anytype) @TypeOf(struct_type) {
                                                     ^~~~~~~~~~~
zig-cache/o/a6f5dfe6a9d692c7d0fdbc4ef13c56d8/cimport.zig:946:110: error: use of undeclared identifier 'struct_headname'
pub inline fn TAILQ_LAST(head: anytype, headname: anytype) @TypeOf(@import("std").zig.c_translation.cast([*c]struct_headname, head.*.tqh_last).*.tqh_last.*) {
                                                                                                             ^~~~~~~~~~~~~~~
zig-cache/o/a6f5dfe6a9d692c7d0fdbc4ef13c56d8/cimport.zig:954:125: error: use of undeclared identifier 'struct_headname'
pub inline fn TAILQ_PREV(elm: anytype, headname: anytype, field: anytype) @TypeOf(@import("std").zig.c_translation.cast([*c]struct_headname, elm.*.field.tqe_prev).*.tqh_last.
*) {
                                                                                                                            ^~~~~~~~~~~~~~~
```

The aforementioned xlsxwriter seems to borrow some of that same code (at least the `TAILQ` names sound familiar)

**Comment by kassane** [comment]
> This happens with any FreeBSD system header that pulls `queue.h` :(
> The aforementioned xlsxwriter seems to borrow some of that same code (at least the `TAILQ` names sound familiar)

try:
```zig
@cImport({
    @cDefine("struct_type", "");
    @cInclude("queue.h");
});
```
similar on: https://github.com/kassane/zig-xlsxwriter/blob/c32228725fb36a05e8bae52857e926d209cb0530/src/xlsxwriter.zig#L2C1-L5C4

**Comment by mlugg** [comment]
Was resolved by 0fa8cf4
